source/patterns/@aws-solutions-constructs/core/lib/opensearch-helper.ts

```diff
@@ -0,0 +1,306 @@
+/**
+ *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import * as opensearch from 'aws-cdk-lib/aws-opensearchservice';
+import { DefaultOpenSearchCfnDomainProps } from './opensearch-defaults';
+import { consolidateProps, addCfnSuppressRules } from './utils';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as cdk from 'aws-cdk-lib';
+import * as cloudwatch from 'aws-cdk-lib/aws-cloudwatch';
+import * as cognito from 'aws-cdk-lib/aws-cognito';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+// Note: To ensure CDKv2 compatibility, keep the import statement for Construct separate
+import { Construct } from 'constructs';
+
+const MAXIMUM_AZS_IN_OPENSEARCH_DOMAIN = 3;
+
+export interface BuildOpenSearchProps {
+  readonly identitypool: cognito.CfnIdentityPool;
+  readonly userpool: cognito.UserPool;
+  readonly cognitoAuthorizedRoleARN: string;
+  readonly serviceRoleARN?: string;
+  readonly vpc?: ec2.IVpc;
+  readonly openSearchDomainName: string;
+  readonly clientDomainProps?: opensearch.CfnDomainProps,
+  readonly securityGroupIds?: string[]
+}
+
+export function buildOpenSearch(scope: Construct, props: BuildOpenSearchProps): [opensearch.CfnDomain, iam.Role] {
+  let subnetIds: string[] = [];
+  const constructDrivenProps: any = {};
+
+  // Setup the IAM Role & policy for the OpenSearch Service to configure Cognito User pool and Identity pool
+  const cognitoKibanaConfigureRole = createKibanaCognitoRole(scope, props.userpool, props.identitypool, props.openSearchDomainName);
+
+  if (props.vpc) {
+    subnetIds = retrievePrivateSubnetIds(props.vpc);
```

Comment:
Replicating this function verbatim from elasticsearch-helper? C'mon - that's just silly.

Perhaps we can find a better place to move this function and share it (perhaps vpc-helper.ts?). References to the search engines should be removed from the function's code.